graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/functions/conversion/ListConversion.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.plugins.pipelineprocessor.functions.conversion;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.graylog.plugins.pipelineprocessor.EvaluationContext;
+import org.graylog.plugins.pipelineprocessor.ast.functions.FunctionArgs;
+import org.graylog.plugins.pipelineprocessor.ast.functions.FunctionDescriptor;
+import org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor;
+import org.graylog.plugins.pipelineprocessor.rulebuilder.RuleBuilderFunctionGroup;
+
+import java.util.Collections;
+import java.util.List;
+
+import static com.google.common.collect.ImmutableList.of;
+import static org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor.object;
+
+public class ListConversion extends AbstractConversion<List> {
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    public static final String NAME = "to_list";
+    private static final String VALUE = "value";
+
+    private final ParameterDescriptor<Object, Object> valueParam;
+
+
+    public ListConversion() {
+        this.valueParam = object(VALUE).ruleBuilderVariable().description("List-like value to convert").build();
+    }
+
+    @Override
+    public List evaluate(FunctionArgs args, EvaluationContext context) {
+        final Object value = valueParam.required(args, context);
+
+        if (value == null) {
+            return Boolean.TRUE.equals(defaultToNull(args, context)) ? null : Collections.emptyList();
+        } else if (value instanceof List) {
+            return (List) value;
+        } else if (value instanceof JsonNode) {
+            final JsonNode jsonNode = (JsonNode) value;
+            return MAPPER.convertValue(jsonNode, List.class);
+        } else {
+            return Collections.emptyList();
+        }
```

Comment:
What do you think of replacing the if-stmt with switch?
```suggestion
        return switch (value) {
            case null -> Boolean.TRUE.equals(defaultToNull(args, context)) ? null : Collections.emptyList();
            case List list -> list;
            case JsonNode jsonNode -> MAPPER.convertValue(jsonNode, List.class);
            default -> Collections.emptyList();
        };
```

Reply:
I like it - didn't know you could perform a typecast like that in the case clause. Very nifty.